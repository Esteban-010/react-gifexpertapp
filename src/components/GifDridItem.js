import React from 'react'
import PropTypes from 'prop-types';

export const GifDridItem = ({ title, url }) => {
    

    
    return (
        <div className="card animate__animated animate__fadeIn" >
            <img src={ url } alt={ title }/>
            <p>{ title }</p>
        </div>
    )
}

GifDridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
