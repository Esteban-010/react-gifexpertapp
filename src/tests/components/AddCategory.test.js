import React from 'react'
import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'

describe('Pruebas en AddCategory.js', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    })
    
    test('debe de mostrarse correctamente', () => {
       expect(wrapper).toMatchSnapshot();
    })
    
    test('debe cambiar la caja de texto', () => {
        const value = 'Hola mundo';
        
        wrapper.find('input').simulate('change', { target: { value } });

        expect(wrapper.find('input').prop('value').trim()).toBe(value);
    })
    
    test('No debe de postear la informacion del submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('debe llamar al setCategories y limpiar la caja de texto', () => {
        
        const value = 'Hola mundo';
        
        wrapper.find('input').simulate('change', { target: { value } });
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).toHaveBeenCalled();

        expect(wrapper.find('input').prop('value').trim()).toBe('');
    })
    
    
})
