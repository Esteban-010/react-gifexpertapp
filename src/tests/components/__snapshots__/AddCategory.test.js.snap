// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory.js debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p
    style={
      Object {
        "display": "none",
      }
    }
  />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
